### DISCUSSION TITLE
Trying to take image tiles and combine them together.

### DISCUSSION THREAD

**Discussion by GoldenFurredFox** [ImageMagick/ImageMagick#6511] [Help]
### ImageMagick version

7.1.0-39

### Operating system, version and so on

Microsoft Windows 10 Home - Version 10.0.19045 Build 19045 - 64 bit

### Description

I found a [discussion post](https://github.com/ImageMagick/ImageMagick/discussions/3449) here before, which suggested using convert to combine tiles as gifs. It worked flawlessly, and I was able to create this monstrosity (seen below) as a banner for a forum site.

![The monstrosity in question.](https://i.imgur.com/eo1qFAr.gif)

The code snippet he'd shared essentially went like this: 

You have the original width and height of both images. For simplicity sake, let's just say the width and height of both images are 128. You'd run the below code, and it'd slap both the gifs together, easy peasy, making a new file that is a 256x128 gif.

magick convert .\file1.gif -repage 256x128 -coalesce null: ( .\file2.gif -coalesce ) -geometry +128+0 -layers Composite .\file3.gif

This file, step by step, takes file1, and repages a temporary copy to extend the canvas to 256x128, adds file2 at position x128 y0, and outputs the temporary combination of file1 and file2 as file3.

It works fine for GIFs, but now I want to do it for PNGs. A friend of a friend recently sent me a zip file of three thousand six hundred 16x16 images all named based on their position (40 right and 40 down would be 40AN, for example - the format often used in spreadsheet apps) and said he'd order me a brand new Steam Deck by mail to a nearby PO box if I could combine them all in less than 6 hours to form a 1280x720 image, but he didn't say I couldn't write code to do it for me, so I was gonna make a batch prompt to do it in mere seconds/minutes and blow his mind. (Guy is like a mini Mr. Beast in terms of content - filthy rich and donates often, but also extremely chill when not actively making content.)

**Comment by GeeMack** [comment]
If the file names are in the correct sequential order, a command to assemble the 3600 images could be as simple as this...

    magick input*.png +append +repage -crop 1280x -append +repage result.png

That would read all the images into the command, "+append" them into a single horizontal row, crop that row into 45 units of 1280 pixels wide, and "-append" them vertically to create the 1280x720 result.

It takes about 3 seconds.

**Reply by GoldenFurredFox** [reply]
Would I need to substitute out any text in that command? Despite what I said, I am a newbie to both making batch files and using ImageMagick.

**Reply by GoldenFurredFox** [reply]
Oh, and also, how and where does that get the 45 from?

**Reply by GeeMack** [reply]
That command above would assemble the image if the filenames were, for  example, AA001 ... AA080, AB001 ..AB080, etc., using the way the OS reads them into the command with the wildcard "*". I don't understand what sort of filenames you're starting with, but some adjustment in the naming convention might be all it takes to make the ImageMagick command extremely simple.

**Reply by GoldenFurredFox** [reply]
Sorry for not replying back sooner. Cells horizontally all share a number and have different letters, while ones aligned vertically have the same letter and incrementing numbers. The letters go by Base26, while the numbers remain Base10, hence why I said that the cell 40 across and 40 down would be cell 40AN - N is the 14th letter of the alphabet, but since it incremented the entire alphabet once before reaching 40, the next digit place became A.

**Reply by GoldenFurredFox** [reply]
I was going to use 32 as an example initially, but in the Base26 alphabetical counting system, 32 is AF, and saying something is 32AF seems weird out of context.

**Comment by snibgo** [comment]
> 40 right and 40 down would be 40AN, for example

What is the name of the image on the same row, to the right of this one? The image at 41 right and 40 down? I suppose it would be either 41AN or 40AO.

Geemack's solution depends on the filenames being in lexicographical ("dictionary") order. So when the they are listed in order, is the top-left image the first in the list? And the image next to the top-left image is the second in the list? And so on.

**Reply by GoldenFurredFox** [reply]
Yes, the top left would be 1A, the one directly to the right would be 1B, and so on and so forth. Like how tiles are laid out in a spreadsheet. That means that the one 41 right and 40 down would be 40AO.

... though, the window for me to complete this project is long since overdue, sadly.
